typedjsonrpc/registry.py

```diff
@@ -107,9 +107,12 @@ def _wrapped():
 
         return self._handle_exceptions(_wrapped, is_notification, self._get_id_if_known(msg))
 
-    def _handle_exceptions(self, method, is_notification=False, msg_id=None):
+    def _handle_exceptions(self, method, is_notification=False, msg_id=None, params=None):
```

Comment:
I have a preference to keep this method simpler and have the method which is passed in capture the parameters inside it. That way we don't have to have an if-else statement in here which isn't really relevant to the exception handling code